unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,301 @@
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await UniTask.WaitUntil(() =>
+                    socialClient != null,
+                PlayerLoopTiming.Update,
+                cancellationToken);
+
+            // start listening to streams
+            // await UniTask.WhenAny(this.ListenToFriendEvents(cancellationToken));
+        }
+
+        public async UniTask<FriendshipInitializationMessage> InitializeFriendshipsInformation(CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+
+            await UniTask.WaitUntil(() =>
+                    socialClient != null,
+                PlayerLoopTiming.Update,
+                cancellationToken);
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var ownUserProfile = UserProfile.GetOwnUserProfile();
+
+            await UniTask.WaitUntil(() =>
+                    !string.IsNullOrEmpty(ownUserProfile.userId) && !string.IsNullOrEmpty(this.accessToken),
+                PlayerLoopTiming.Update,
+                cancellationToken);
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            var friendsStream = socialClient.GetFriends(new Payload() { SynapseToken = this.accessToken });
+
+            await foreach (var friends in friendsStream.WithCancellation(cancellationToken))
+            {
+                foreach (var friend in friends.Users_) { OnFriendAdded?.Invoke(new UserStatus { userId = friend.Address }); }
+            }
+
+            cancellationToken.ThrowIfCancellationRequested();
+
+            // var requestEvents = await rpc.Social().GetRequestEvents(new Empty());
+            //
+            // foreach (var friendRequest in requestEvents.Incoming.Items)
+            // {
+            //     OnIncomingFriendRequestAdded?.Invoke(new FriendRequest(
+            //         GetFriendRequestId(friendRequest.User.Address, friendRequest.CreatedAt), friendRequest.CreatedAt, friendRequest.User.Address, ownUserProfile.userId, friendRequest.Message));
+            // }
+            //
+            // foreach (var friendRequest in requestEvents.Outgoing.Items)
+            // {
+            //     OnOutgoingFriendRequestAdded?.Invoke(new FriendRequest(
+            //         GetFriendRequestId(friendRequest.User.Address, friendRequest.CreatedAt), friendRequest.CreatedAt, ownUserProfile.userId, friendRequest.User.Address, friendRequest.Message));
+            // }
+
+            return new FriendshipInitializationMessage()
+            {
+                // totalReceivedRequests = requestEvents.Incoming.Items.Count,
+                totalReceivedRequests = 0
+            };
+        }
+
+        public async UniTask ListenToFriendEvents(CancellationToken cancellationToken = default)
+        {
+            var ownUserProfile = UserProfile.GetOwnUserProfile();
```

Comment:
instead of referring to `UserProfile.GetOwnUserProfile();` you should inject a `IUserProfileBridge` and then use: `userProfileBridge.GetOwn()`